### ISSUE TITLE
GPS info during OpenSfM

### ISSUE THREAD

**Issue by dakotabenjamin** [edgarriba/OpenDroneMap#16]
I was running the process with odm_data/seneca, which has GPS coords in the exif on all the photos (I double-checked), but I got this warning: 

```
WARNING:__main__:Not all images have GPS info. Disabling matching_gps_neighbors.
```

I ran it on the master branch and did not run across this issue.

**Comment by edgarriba** [comment]
Not sure but I guess it's that flag.
https://github.com/edgarriba/OpenDroneMap/blob/cmake/scripts/opensfm.py#L59

Anyways it's only a warning that's tells you what you said. Any consideration @paulinus ?

**Comment by dakotabenjamin** [comment]
I'm wondering if it has to do with https://github.com/OpenDroneMap/OpenDroneMap/commit/1cf79b64fffc0cd52198229b4099f6aad496cc5c

**Comment by dakotabenjamin** [comment]
What is happening in the declare_params() call? https://github.com/edgarriba/OpenDroneMap/blob/cmake/scripts/opensfm.py#L9

```
    def declare_params(self, params):
        params.declare("use_exif_size", "The application arguments.", False)
        params.declare("feature_process_size", "The application arguments.", False)
        params.declare("feature_min_frames", "The application arguments.", 0)
        params.declare("processes", "The application arguments.", 0)
        params.declare("matching_gps_neighbors", "The application arguments.", 0)
```

Is the last arg in params.declare supposed to be the default?

**Comment by edgarriba** [comment]
https://github.com/edgarriba/OpenDroneMap/blob/cmake/scripts/odm_app.py#L41

takes the value in 'matcher_k' which right I'm not sure if it's correct

**Comment by dakotabenjamin** [comment]
I'm currently updating those arguments to matcher_neighbors and matcher_distance. We'll see if it changes anything!

**Comment by edgarriba** [comment]
nice!

**Comment by dakotabenjamin** [comment]
well I'm not getting anywhere with the matcher-distance args, but I did find the source of the warning. Looking into it now:
https://github.com/mapillary/OpenSfM/blob/odm-2/bin/match_features#L48

**Comment by dakotabenjamin** [comment]
Yeah I installed OpenSfM separately and ran bin/run_all, had the same warning. The config.yaml file has matching_neighbors, so `all(map(has_gps_info, exifs))` must be returning false. ~~My guess is the new directory design means that opensfm is not getting the exif info from the images.~~ The image lists file uses absolute references to the images, I don't see how that could be the problem.

**Comment by paulinus** [comment]
There was a bug in that line checking for gps info. Just fixed that in OpenSfM odm-2 branch. Sorry for being slow here.

Btw, what will be the procedure for updating opensfm when using releases, instead of tracking a branch.

**Comment by dakotabenjamin** [comment]
That's a good question. How often do you expect OpenSfM to update such that it affects ODM?

**Comment by dakotabenjamin** [comment]
After that fix OpenSfM seems to be running at speeds like the master branch. I'll run some tests and PR an update.
